src/services/user.service.js

```diff
@@ -0,0 +1,44 @@
+const { data } = require('../data');
+
+const getAll = () => {
+  return data.users;
+};
+
+const getById = (id) => {
+  return data.users.find((user) => user.id === id) || null;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getById(id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
+
+const createUser = (name) => {
+  const user = {
+    name,
+    id: data.nextId++,
+  };
+
+  data.users.push(user);
+
+  return user;
+};
+
+const deleteUser = (id) => {
+  const initialLength = data.users.length;
+
+  data.users = data.users.filter((user) => user.id !== id);
+
+  return data.users.length < initialLength;
```

Comment:
In the `deleteUser` function, you are directly modifying the `data` array by filtering out the user to be deleted. This could lead to unintended side effects in other parts of the code that are using the `data` array. Consider returning a new array without the deleted user instead of modifying the existing one. Also, you should handle the case when the user is not found.